### ISSUE TITLE
Installer shows an extension (vsix) issue on deistallation (latest Nightly)

### ISSUE THREAD

**Issue by dan31** [Starcounter/Starcounter#3486]
Moving from SC version from March 24 to a current one. (It was pnext installed before, now it is develop, I don't know whether it matters or not, only mention for report).
This one is reported during uninstallation:
<img width="623" alt="pngimg" src="https://cloud.githubusercontent.com/assets/7162535/14432381/3b099ffc-000a-11e6-95ed-52d3a6b308db.png">
Investigating this further on, I paste and execute this command:
``` C:\Users\mbp>"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\VSIXInstaller.exe" /skuName:Community /skuVersion:14.0 /uninstall:Starcounter.VS14.96DB7E43-995A-4754-BDDB-7E37EDB2281B ```
Which results in 
![image](https://cloud.githubusercontent.com/assets/7162535/14432535/e5932af6-000a-11e6-9afe-2ca386fb336a.png)
That is indeed true, since I have no Community edition, I have Enterprise (they don't live together, installing Enterprise automatically subsumes Community). However, trying this with sku:Enterprise doesn't appear to work either.
And only when I removed the SC extension from VS folder like this (moved from right to left):
![image](https://cloud.githubusercontent.com/assets/7162535/14433268/f05720a2-000d-11e6-9e1f-08950aa96714.png)
The new installer actually worked.

**Comment by dan31** [comment]
After deinstalling the installed one I get similar message, but different options reported in the window:
![image](https://cloud.githubusercontent.com/assets/7162535/14434876/7a0ba6d6-0015-11e6-94d0-88ee2614632a.png)

**Comment by dan31** [comment]
One more note: I have VS 2015 with Update 2 (Released March 30 2016).

**Comment by ubbeK** [comment]
I got the same error message

> Error occurred during uninstall:
System.Exception: ScErrVSIXEngineFailed (SCERR11030): The VSIX installer engine failed to install or uninstall the deployment package. Process exit code=1002, Engine path=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\VSIXInstaller.exe, Arguments=/quiet /skuName:Community /skuVersion:14.0 /uninstall:Starcounter.VS14.96DB7E43-995A-4754-BDDB-7E37EDB2281B
Version: 2.1.1764.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR11030.
   at Starcounter.InstallerEngine.UninstallEngine.UninstallAllComponents()
   at Starcounter.InstallerEngine.UninstallEngine.UninstallStarcounter(Boolean cleanup)
   at Starcounter.InstallerEngine.InstallerMain.StarcounterSetup(String[] args, String binariesPath, String setupConfigFile, EventHandler`1 progressCallback, EventHandler`1 messageboxCallback)

I am using Windows 10 Pro and Visual Studio 2015 Community Update 2.

**Comment by dan31** [comment]
Seems to me that the reason is having Update 2 installed. More people will eventually have this problem over next 1-2 months.

**Comment by dan31** [comment]
Still have to manually delete a plugin (tried Stable 2.2.1811)

**Comment by per-samuelsson** [comment]
We need to get on top of this.

@dan31, do you know if you changed something in VS between first installing the extension, and later trying to uninstall it? I'm thinking that could be a reason....

**Comment by per-samuelsson** [comment]
FWIW, here's a resource that shows VSIXInstaller exit codes:
http://stackoverflow.com/questions/21531596/vsixinstaller-exe-exit-code-documentation/28212173#28212173

So errors here are consistent: the VSIX installer consider the given extension not installed (i.e. `1002`). We just need to figure out why.

**Comment by per-samuelsson** [comment]
@dan31 @ubbeK 

Can you confirm your workflow have been

1. Having Community (< Update 2).
2. Installing Starcounter extension
3. Altering VS - in @dan31's case, upgrading to Enterprise; in @ubbeK's case, updating to Commnity Update 2.
4. Uninstalling the extension (failing)

**Comment by per-samuelsson** [comment]
I think the installer is a bit too simplistic for VS 2015, just considering "Commnity" edition to be some common denominator when installing (and uninstalling). We probably have to add some logic to detect what exact editions are installed and act on that.

@dan31 @ubbeK 

Could you assist and bring down this new utility I've created that reveal some info on editions on a given system. Just clone and build the single solution in https://github.com/per-samuelsson/vseditions, and then hit F5, and editions found should be displayed in the console. I'm particularly interested in what would be displayed on @ubbeK's box. Thanks in advance.

**Comment by ubbeK** [comment]
@per-samuelsson: I cannot confirm that I ever had a lesser Update pack than Update 2. Altought, I did install Starcounter when using Windows 7, then upgraded to Windows 10 and then failing when uninstalling Starcounter. I'll try out your solution.

**Comment by per-samuelsson** [comment]
> I cannot confirm that I ever had a lesser Update pack than Update 2. Altought, I did install Starcounter when using Windows 7, then upgraded to Windows 10 and then failing when uninstalling Starcounter.

Aha, ok, thanks. That information might be usable.

> I'll try out your solution.

Thanks! I'm just reviewing it, to see if we can somehow better reflect all editions actually installed in a more accurate manner, but I still haven't found anything. Let's see what your results provide. It can be that you will see nothing, which would indicate that "Community Update 2" is considered a unique edition.

**Comment by per-samuelsson** [comment]
> I'll try out your solution.

Notice I just pushed a new versions that display more diagnostics: https://github.com/per-samuelsson/vseditions. If you can pull and rerun that would be of great help!

**Comment by ubbeK** [comment]
@per-samuelsson: I ran your latest version with the result:

> Keys we know about:
> VS 2015/14.0: Community
> Found keys per version:
> VS 2015/14.0:
>   BuildNumber
>   community
>   JSLS_MSI
>   JSPS_MSI
>   SecondaryInstaller

**Comment by per-samuelsson** [comment]
> @per-samuelsson: I ran your latest version with the result:

Thanks a lot.

Those result are a bit surprising. They are similar to those I have on my box, and I have no problem uninstalling, but I'm on Update 1. I'll continue to investigate and try figure out some way to solve this.

**Comment by k-rus** [comment]
I have the same problem with uninstallation. I have installation of Starcounter from local build of pnext. It happened that Update 2 was installed yesterday. Today I need to uninstall Starcounter due to #3513. During uninstall I get the same exception as above:

```
Error occurred during uninstall:
System.Exception: ScErrVSIXEngineFailed (SCERR11030): The VSIX installer engine failed to install or uninstall the deployment package. Process exit code=1002, Engine path=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\VSIXInstaller.exe, Arguments=/quiet /skuName:Community /skuVersion:14.0 /uninstall:Starcounter.VS14.96DB7E43-995A-4754-BDDB-7E37EDB2281B
Version: 2.0.0.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR11030.
   at Starcounter.InstallerEngine.UninstallEngine.UninstallAllComponents() in C:\GitRepositories\Starcounter\Level1\src\Starcounter.Installer\Starcounter.InstallerEngine\UninstallEngine.cs:line 228
   at Starcounter.InstallerEngine.UninstallEngine.UninstallStarcounter(Boolean cleanup) in C:\GitRepositories\Starcounter\Level1\src\Starcounter.Installer\Starcounter.InstallerEngine\UninstallEngine.cs:line 289
   at Starcounter.InstallerEngine.InstallerMain.StarcounterSetup(String[] args, String binariesPath, String setupConfigFile, EventHandler`1 progressCallback, EventHandler`1 messageboxCallback) in C:\GitRepositories\Starcounter\Level1\src\Starcounter.Installer\Starcounter.InstallerEngine\InstallerMain.cs:line 742

Do you want to completely cleanup your system from Starcounter?
```
I run cleanup and it was executed successfully.
Installing Starcounter on latest source code was successful.

**Comment by per-samuelsson** [comment]
Thanks for sharing that @k-rus. It seems that @dan31's early guess is spot on - this is something that is caused by introducing Update 2. I'm going to try that loop myself, and see if I can debug what invocation actually change. I have a vague suspicion, let's see how it turns out.

> Installing Starcounter on latest source code was successful.

You mean installing again, after you have ran the cleanup? Have you tried again to uninstall after that? (You don't need to, I'll try that cycle myself, just curious).

**Comment by k-rus** [comment]
> You mean installing again, after you have ran the cleanup? Have you tried again to uninstall after that? (You don't need to, I'll try that cycle myself, just curious).

Yes, I installed Starcounter after the clean up and I haven't tried to uninstall it.

**Comment by per-samuelsson** [comment]
> Yes, I installed Starcounter after the clean up and I haven't tried to uninstall it.

Thanks.

@dan31, would be very interesting to see the results of running my new utility on that machine of yours, where you have enterprise installed:

> Could you assist and bring down this new utility I've created that reveal some info on editions on a given system. Just clone and build the single solution in https://github.com/per-samuelsson/vseditions, and then hit F5, and editions found should be displayed in the console.

**Comment by per-samuelsson** [comment]
> I have a vague suspicion, let's see how it turns out.

From the release notes of Update 2, I found this:

> Reduced the time for Visual Studio to start when extensions are installed.

I think that change might be the one that broke our uninstaller, since we keep a workaround (invoking VSIXInstaller twice) to make sure the uninstall is accurately reporting a correct value when the installer later ask if it's uninstalled (I didn't previously).

**Comment by per-samuelsson** [comment]
I can now confirm that once I've installed Update 2, I experience the same thing, and that the theory I commented on above holds true - the VSIX installer has changed the exit code the second time it's invoked (it's now 1002, while it was previously 2003, something we handled).

If we are lucky, it might be as easy as taking height for that code too, and it will then work with both Update 1 and 2. However, I'm still not confident that solves everything, since I still have no info on how things work out on a machine with Enterprise only (like @dan31's machine), and I need those stats before I can trigger a new release.

**Comment by per-samuelsson** [comment]
> If we are lucky, it might be as easy as taking height for that code too, and it will then work with both Update 1 and 2.

I've implemented that fix and tested a few installs/uninstalls on my box, and things now seem to work as expected. I haven't merged it into any main branch yet though, because ...

> However, I'm still not confident that solves everything, since I still have no info on how things work out on a machine with Enterprise only (like @dan31's machine), and I need those stats before I can trigger a new release.

... I want @dan31 to confirm it works on his env before that, and therefore I'm currently running a personal build with the fix, and generating a new installer based on that. I'll give you a link to it, once the build server succeed with it, and once you confirm it works, I'll merge it into the main branch.

**Comment by per-samuelsson** [comment]
> [...] therefore I'm currently running a personal build with the fix, and generating a new installer based on that. I'll give you a link to it,

[Here](https://scbuildserver/repository/download/Starcounter_DevelopDaily/2894:id/Starcounter-DevelopDailyBuilds-2.2.1819-Setup.exe) you go @dan31, please try that one, and let me know if you can run a successful install/uninstall cycle with it.

**Comment by per-samuelsson** [comment]
Decided on a bit more optimistic approach, and merged the fix into **develop**, in https://github.com/Starcounter/Starcounter/commit/8295a380a1c67f24997505daf56dc2171feffb21. 

If @dan31 later can confirm it works with that installer I provided above, we know that the fix get part of tonights nightly. Once that happen, I'll also update the changelog and merge it into **pnext**.

**Comment by dan31** [comment]
@per-samuelsson interesting output!
```
Keys we know about:
VS 2013/12.0: Professional
VS 2015/14.0: Professional
VS 2015/14.0: Community
VS 2015/14.0: Enterprise
Found keys per version:
VS 2010/10.0:
  BuildNumber
VS 2013/12.0:
  BuildNumber
  Pro
  VSTD
  VSTS
VS 2015/14.0:
  BuildNumber
  community
  enterprise
  JSLS_MSI
  JSPS_MSI
  MDD
  Pro
  SecondaryInstaller
  VSTD
  VSTS
```

So it thinks I have both Community and Enterprise. However I know that I have no means to run Community, only Enterprise. When I was installing, I first had Community and then installed Enterprise. I simply wanted to have them both. Don't see it is actually possible.

One more cool thing. I also have this:

![image](https://cloud.githubusercontent.com/assets/7162535/14651818/d4323e1a-0671-11e6-8854-d31e457c3777.png)

Yes, Express exists! The difference is licensing terms, and perhaps some limitations. For instance, this might the case that Express doesn't support plugins. But I'm not sure.

**Comment by per-samuelsson** [comment]
> So it thinks I have both Community and Enterprise. However I know that I have no means to run Community, only Enterprise. When I was installing, I first had Community and then installed Enterprise. I simply wanted to have them both. Don't see it is actually possible.

Yes, that's what I expected. And it's good, because then I think the current implementation will work. When we install the extension, we always specify "Community" when we pass it to the VSIX installer, and I think that will work, since all other versions are built on top of that. As an example, if you install Enterprise, I think its "Community + extras" underneath.

Please try that installer I provided above to see if the full cycle work on your box with this fix I provided. If that works, we are in good shape. If we want to be 100%, we should test on a system that install `Enterprise` right from the start though.

(And oh, as a friendly advice, you really should drop some old stuff from time to time. Traces of VS 2010 (!!) still, I mean come on :wink:)

**Comment by per-samuelsson** [comment]
> For instance, this might the case that Express doesn't support plugins. But I'm not sure.

Yes, Express never supported extensions, so we could never target it. Now Community does though, and Express is not more an active player.

**Comment by dan31** [comment]
> (And oh, as a friendly advice, you really should drop some old stuff from time to time. Traces of VS 2010 (!!) still, I mean come on :wink:)

The truth is that I never had it! This trace belongs to VS-2012 I believe.

**Comment by per-samuelsson** [comment]
> The truth is that I never had it! This trace belongs to VS-2012 I believe.

Hmm, that's weird, but maybe they design their product updates such as they seem to do with editions, so that 2012 is a release that is 2010 + more, and it comes part of that. No matter what, I guess it's not very important right now. I just want to make sure any problems are actually fixed.

**Comment by per-samuelsson** [comment]
@ubbeK @dan31 

If you guys could verify this issue is actually fixed by running a complete install/uninstall cycle again using our newest nightly - [2.2.1838](http://downloads.starcounter.com/download/Starcounter/DevelopNightlyBuilds/2.2.1838) - that would be really great. I'd like to keep this one open until I'm sure you actually confirm it's fixed.

**Comment by ubbeK** [comment]
> If you guys could verify this issue is actually fixed by running a complete install/uninstall cycle again using our newest nightly - 2.2.1838 - that would be really great. I'd like to keep this one open until I'm sure you actually confirm it's fixed.

@per-samuelsson: I still got the same error when uninstalling from version [2.1.1764](http://downloads.starcounter.com/download/Starcounter/DevelopNightlyBuilds/2.1.1764) to the newest version you mentioned. But there is no problem when i reinstall it from that point on (2.2.1838 to same verstion again).

**Comment by aloscha** [comment]
Maybe I describe my situation, which is really strange.

I modify Chatter example of Starconter and accidentally make some wrong name in class definition. Then Sc go creazy and write down, that I couldn't use namespace SC (no reference). Next I try to uninstall and install again. What I get is the error in first post. I couldn't uninstall it. I try install 5 version back and stable version and have the same results.

Only thing, which helps me is to open working project with SC (I use fresh version of KitchenSink) and then uninstall it without any problems and install again stable version.

**Comment by dan31** [comment]
@aloscha thanks for reporting, can you be more specific (description of changes you made, code samples, exact output message and so on) and open it as a separate issue?

**Comment by per-samuelsson** [comment]
> @per-samuelsson: I still got the same error when uninstalling from version 2.1.1764 to the newest version you mentioned. 

Yeah, that's unfortunately nothing we can do anything about, since uninstalling always is done using the same installer as was used to install. And all installers prior to yesterdays fix will show this symptom if Update 2 have been applied. :cry: 
 
> But there is no problem when i reinstall it from that point on (2.2.1838 to same verstion again).

Then it works on your machine, and mine. Good at least. I'm still awaiting @dan31's feedback, on an Enterprise machine. 

Thanks a lot for all the feedback!

**Comment by per-samuelsson** [comment]
@aloscha 

Sorry you are also experiencing this. It's on us of course, but we couldn't foresee MS would change how the underlying tool (`VSIXInstaller.exe`) works within an update, if that could be something of an excuse.

Regarding this:

> Only thing, which helps me is to open working project with SC (I use fresh version of KitchenSink) and then uninstall it without any problems and install again stable version.

That's a really important thing you mention, and I'm instantly going to reinstall one of the older, failing Starcounter versions and see if I can get that to work. If it does, which I think it does, then I we have a workaround for this error, which would be great until everyone has upgraded. You shouldn't even have to open the project. Just open a new VS instance, go to `Tools | Extensions and updates ...` and then find Starcounter extension and hit `Uninstall`. Uninstalling it like that probably is aligned with the new changes they've introduced in Update 2.

I'll get back once I've tested that.

**Comment by bigwad** [comment]
>Yeah, that's unfortunately nothing we can do anything about, since uninstalling always is done using the same installer as was used to install. And all installers prior to yesterdays fix will show this symptom if Update 2 have been applied. :cry:

Actually, there are some technics to overcome it. Though it's complicated and perhaps an overkill  in our situation. Anyway, I'm aware of two options:
1. Legitimate. Fix old buggy package and distribute it along with the new one. On upgrade, you should at first reinstall old package using REINSTALLMODE=v option. It will substitute buggy package in Windows installer cache with the fixed one. Then just proceed with upgrade to your new package.
2. Insane. Add special custom action in the new package that will find and edit buggy package in Windows installer cache. Say, it can remove failing custom actions actions from appropriate table.

This just FYI. No action required :smile:

**Comment by per-samuelsson** [comment]
Wow, thanks for sharing that @bigwad. However,

1. does that apply to our scenario? We're not using any standard Windows installer, but rather a custom installer written by us. But maybe the hooking in occurs before that in some way.
2. I'm not sure we still want to do that, considering the time it would take and that the problem experienced is o/c not at all something we want our customers to face, but it's neither something that like risk data or projects or apps or whatever. I know you you shared not to push such change, I'm just giving my view on it too.

**Comment by bigwad** [comment]
Ok, this is game changer. If we don't bring standard msi package, then both options above are irrelevant.
Then it's our own decision, what to do on uninstall during upgrade and new package is free to do anything: 
-replace old module with the fixed new one before starting uninstall OR
-call some other module to perform uninstall OR
-change a system in a way that make old module work OR
-don't perform uninstall at all OR
-whatever...

**Comment by atifwaqar** [comment]
I installed SC 2.1.177 some days ago. I have VS Community 2015 installed. 
Now i tried to upgrade my SC to latest Stable version, in the uninstall process the setup uninstalled : 
1. SC
2. VS 2013 component

But it gave exception on uninstalling VS 2015 component. 

as  @per-samuelsson said :

> Yeah, that's unfortunately nothing we can do anything about, since uninstalling always is done using the same installer as was used to install. And all installers prior to yesterdays fix will show this symptom if Update 2 have been applied. :cry:

So i used the 2.1.177 version to uninstall again. but still the same, setup fails to uninstall VS 2015 component. 
I tried to repair my VS 2015, Restarted my machine, tried to uninstall/install again, but the exception is still there.

**Comment by atifwaqar** [comment]
I have this issue on 2 live servers. The only way to fix this quickly was to install 2.1.177 again. (by adding components).
But i still need to upgrade SC Version on the servers.

**Comment by atifwaqar** [comment]
Since these 2 were my live servers, I had to find some way to make them live again, so I uninstalled VS 2015 and SC from both servers, Deleted Starcounter folder. reinstalled SC Version 2.1.1742 which was previously running on both server. And now everything works fine.

There was 1 issue when i installed SC, As soon as I ran the installer, a dialogue box appeared to install `Microsoft Visual C++ 2015 Redistributable`, But it was already installed on my server, I clicked yes anyway and I got an error window, I could not capture the error image, but it was related to `Microsoft Visual C++ 2015 Redistributable`. so I uninstalled it myself. and then SC installer worked fine after it.

**Comment by per-samuelsson** [comment]
@atifwaqar 

Thanks a lot for taking the time to feedback this, and sorry for any inconvinence we've brought to you. Let me consider your experience for a bit and get back on it.

**Comment by per-samuelsson** [comment]
Regarding this:

> There was 1 issue when i installed SC, As soon as I ran the installer, a dialogue box appeared to install `Microsoft Visual C++ 2015 Redistributable`, But it was already installed on my server, I clicked yes anyway and I got an error window, I could not capture the error image, but it was related to `Microsoft Visual C++ 2015 Redistributable`. so I uninstalled it myself. and then SC installer worked fine after it.

I made a separate issue out of it: #3563.

**Comment by per-samuelsson** [comment]
> Let me consider your experience for a bit and get back on it.

@atifwaqar 

Now I've considered everything you wrote in this [this comment](https://github.com/Starcounter/Starcounter/issues/3486#issuecomment-216566651), and when thinking about it, it seem to be the expected behavior (which of course is sad, but that's the bug I was trying to fix in later versions). 

If I understand [this comment](https://github.com/Starcounter/Starcounter/issues/3486#issuecomment-216566651) correctly, your experience was:

1. Uninstalled 2.1.1742 ("[...] I tried to upgrade my SC to latest Stable version" - that will use 2.1.1742 to *uninstall* which I pointed out [here](https://github.com/Starcounter/Starcounter/issues/3486#issuecomment-212400718)). It failed b/c of the bug in the OP.
2. When [1] failed, you tried again to uninstall using 2.1.1742  ("So i used the 2.1.1742 version to uninstall again. but still the same, setup fails to uninstall VS 2015 component."). And that would produce the same result.

As a result, you uninstalled VS and Starcounter and reinstalled 2.1.1742 fresh, which works for you. But now you don't have the latest stable. :cry:

Let me verify exactly in what version the fix I made is part of (I hope latest stable) and get back to you with a guide on how to proceed once you get ready to try again.

**Comment by per-samuelsson** [comment]
> Let me verify exactly in what version the fix I made is part of (I hope latest stable) and get back to you with a guide on how to proceed once you get ready to try again.

Done. And the fix is part of the latest stable, as I hoped for, as hinted by @alemoi [here](
https://github.com/Starcounter/RebelsLounge/issues/16#issuecomment-212109729):

> Ok, I just created a new stable: 2.2.1834 (Apr 19, 2016 9:41:00 PM)
Which contains all the fixes till now, including @per-samuelsson fixes.

That means, when you want to try upgrading those live servers again from 2.1.1742 to new stable (2.2.1834) or a later nightly, do this:

1. Stop all Starcounter processes (hint: `staradmin kill all`)
2. Open a VS 2015 instance (empty instance is fine, you don't need a project).
3. Go to `Tools | Extensions and updates ...`
4. Find **Starcounter VS2015** extension (hint: Search installed...)
5. Hit **Uninstall**
6. Close Visual Studio.
7. Uninstall Starcounter 2.1.1742 (you should now get no error).
8. Install your new version

(As for 7-8, it should be fine just to install the new version as [7], and it will detect that older version need to be uninstalled first, and you still should see no error since you've done 1-6 first).

Please get back to us if we can help you in any other way.

**Comment by aloscha** [comment]
@dan31 

> @aloscha thanks for reporting, can you be more specific (description of changes you made, code samples, exact output message and so on) and open it as a separate issue?

I check that on the last stable version (2.2.1834) and last nightly (2.2.1.1995) and that bug not occur more.

**Comment by per-samuelsson** [comment]
I've fixed this, and there is no more reports that indicate it's still a problem, so I'm closing it.

**Comment by atifwaqar** [comment]
@per-samuelsson  Sorry for the bad news guys. I followed the mentioned steps : 
3. Go to Tools | Extensions and updates ...
4. Find Starcounter VS2015 extension (hint: Search installed...)

but I dont have VS there, and I still have the same issue. I tried to uninstall using the installer that i used to install SC, and also the latest nightly build. But i still have the same issue.

**Comment by atifwaqar** [comment]
Ok I uninstalled VS 2015, and things are back to normal again.

**Comment by per-samuelsson** [comment]
Hm, thats bad. Let's reopen this then. I guess we just need to make those old faulty installations let go somehow.

> [...] but I dont have SC there

Does that mean you don't see it in the menu, when follow those steps? If so, can you run the installer and just choose "Remove components" - does that allow you to select the extension there (or does it say it's not installed)?

Also, can you repeat you setup as it is right now? 

* `star -v` gives you what?
* `Visual Studio | Help | About ...` shows what edition and version?
* Do you have more VS versions installed?

**Comment by atifwaqar** [comment]
> Does that mean you don't see it in the menu, when follow those steps?

Yes. I dont see it in the menu.

> If so, can you run the installer and just choose "Remove components" - does that allow you to select the extension there (or does it say it's not installed)

Yes, it allows me to select the extension there. and it shows that the component is installed, which is weird, and when I hit next, it shows an error (the same one as the first attachment in this post.)

> Also, can you repeat you setup as it is right now?

Sorry, I had to uninstall VS 2015, because it was my live server, and I could not afford to have it down. 

> Do you have more VS versions installed?

Yes, I have VS 2013 installed. and after uninstalling VS 2015, everything works fine now

**Comment by per-samuelsson** [comment]
OK, thanks for the feedback, @atifwaqar.

Then I'm thinking we might want to close this again, although I'm not entirely happy with this:

> Yes, it allows me to select the extension there. and it shows that the component is installed, which is weird, and when I hit next, it shows an error (the same one as the first attachment in this post.)

It's kind of hard to track down though, now that there's no environment, and maybe it was caused by something that was failing in that older version (which in fact had a bug, that we fixed).

Dont hesitate to let me know if you want me to reopen it and/or want some more help.

**Comment by atifwaqar** [comment]
@per-samuelsson You are right, let keep this close. because its not a real issue, i dont think anyone else is gonna regenerate my environment, since I modified a lot of things in panic :P
and now that things are fixed, so i dont think anyone need to spend time on this issue. 

Thanks anyway for the support. I really appreciate it :)

**Comment by dan31** [comment]
Was strike by this again. I can't say when and how this happens, but it happens quite often when I uninstall. Is anyone else experiencing this?
